### ISSUE TITLE
showing different snap shot lists depending on external hdd being connected

### ISSUE THREAD

**Issue by mokafolio** [bit-team/backintime#1016]
Hello,

I am experiencing some weirdness with my setup. I am backing up to an external hdd and it appears to work. The schedule is set to hourly and if the external hdd is connected everything works as expected and the snapshot list in back in time looks correct. If the hdd is not connected though, it shows another list of snapshots which I guess ran while the hdd was not connected. Why is that happening? I'd expect it to simply not take a snapshot if the external hdd is not connected. I am pretty sure its only diffing stuff without copying the backup anywhere in the latter case.

thank you!

**Comment by colinl** [comment]
What do you see if you look in the mount point where the drive would be if it were connected?

**Comment by mokafolio** [comment]
Ah, that exactly is the problem. If the external HDD is not connected, it just uses the folder of the mount point to store the backups. Is there any way to disable that behavior? I'd like it to only create backups if the HDD is connected. Thanks!

**Comment by colinl** [comment]
How do you run BiT?

**Comment by mokafolio** [comment]
I am using the qt client (v. 1.2.0) with basically the default settings.

**Comment by colinl** [comment]
I believe there are hooks in BiT that let you run a script before a backup.  You could check the drive is mounted in there.  I haven't used the scripting but I assume it is documented somewhere.
In the meantime if you empty the mount point folder (make sure the disc is not plugged in!) the it will be pretty obvious that something is wrong as there will be no snapshots shown.

**Comment by mokafolio** [comment]
where would I find those in the gui version?

**Comment by colinl** [comment]
I don't know.  Perhaps someone else can tell us how.

**Comment by mokafolio** [comment]
So I worked around this. Instead of using the mount point directly as the backup location, I created a folder to use that only exists when the extern HDD is mounted. This should work. Would be cool to find another way though! Thank for all the input!

**Comment by buhtzz** [comment]
@mokafolio There is a new forming maintaining team and we do review all issues. Is this problem still relevant for you, can you reproduce it with a newer or the latest release? Did you find a solution?

Tag: Bug, Feedback, Documentation

Here again we have the problem when the destination is a mount point instead of a folder under the mount point.

**Comment by emtiu** [comment]
Yes, I've described the same workaround here: https://github.com/bit-team/backintime/issues/1230#issuecomment-1198018112

This is definitely something we need to look into.

**Comment by buhtz** [comment]
For my info, how to check if mounted on shell:

    $ findmnt --no-headings --options TARGET <path>

Or with Python itself:

    >>> import psutil
    >>> psutil.disk_partitions(all=True)

But the easiest might be this and checking return code:

    $ mountpoint <path>

**Comment by buhtz** [comment]
Hello together,
let us warm it up and coming back to the original use case described.

> I am experiencing some weirdness with my setup. I am backing up to an external hdd and it appears to work. The schedule is set to hourly and if the external hdd is connected everything works as expected and the snapshot list in back in time looks correct. If the hdd is not connected though, it shows another list of snapshots which I guess ran while the hdd was not connected. Why is that happening? I'd expect it to simply not take a snapshot if the external hdd is not connected.

I wondering how BIT can cover this use case. For BIT the destination path does exist in both cases; HDD connected or disconnected. It seems to me that the user himself is responsible for mounting the HDD, not BIT.

So how should BIT know if a backup destination is a mount point or not? Without knowing it it can not do any checking.

Or am I wrong?

**Comment by buhtz** [comment]
Some thoughts and ideas about it. Please correct me if I get something wrong.

1. The use case is relevant only for "local" and "local (encrypted)" profiles. Am I right so far?
2. The profile is scheduled via cron, but should run only if the USB device is connected which means being mounted at the mount point (the backup destination path aka snapshot path).
3. In consequence BIT somehow need to know that the backup destination is a mount point and not a path. Maybe like this:

![Image](https://github.com/user-attachments/assets/1e0b7177-fe77-452d-83d0-db2c079f8bee)

But I don't like that checkbox. To technical. Not all users know what a mount point is. BIT is not targeted for experts. Every use should understand the settings without reading the documentation.
I might hide this option somewhere in "Expert Options" tab.

Another idea was to implement a check: If the path is a mountpoint, does it hide other files and directories behind it, when it is not a mountpoint? In this case BIT could give a warning, that the previous backup might be written to the local filesystem instead to the mountpoint. To my research this is technically not easy and BIT need root permissions for this to check. Seems not to be a good option.

Other idea: BIT is not responsible. But users can use a user-callback script to check it themselves.

I am interested in your opinions.

**Comment by buhtz** [comment]
I really need some user feedback about that "problem" and use case. 🤟 

To my current research there is no way for BIT to know if the backup destination path is (or should be) a mount point or not. The only option I see is the extra checkbox to explicit enable a check.

But as I told I am not convinced and happy about such a checkbox.

In my humble opinion it is a kind of a misuse to setup an external and not always available USB disc as destination and also setup a interval schedule. In best words: It seems to be an edge case. And for cases like this there is the user-callback script where the user can check if the thing is mounted or not.

Even after some days of thinking about it I doubt to handle this as a bug of BIT and tend to close it without any action.

**Comment by buhtz** [comment]
Closing this ticket based on the comment above. Feel free to reopen
if the problem still exists. Thank you for your efforts. If you have
any further questions, ideas or encounter any other issues, please
don't hesitate to let us know.

Best regards,
Christian
